cylc/flow/config.py

```diff
@@ -1723,8 +1723,10 @@ def process_config_env(self):
         os.environ['CYLC_CYCLING_MODE'] = self.cfg['scheduling'][
             'cycling mode']
         # Add workflow bin directory to PATH for workflow and event handlers
-        os.environ['PATH'] = os.pathsep.join([
-            os.path.join(self.fdir, 'bin'), os.environ['PATH']])
+        env_path = [os.path.join(self.fdir, 'bin'), os.environ['PATH']]
+        if self.share_dir is not None:
+            env_path.insert(0, os.path.join(self.share_dir, 'bin'))
```

Comment:
Not a big deal, but it would be cleaner to handle share_dir first, to avoid having to use `list.insert()`.

Reply:
What:
```
        env_path = []
        if self.share_dir is not None:
            env_path.append(os.path.join(self.share_dir, 'bin'))
        env_path.extend([
            os.path.join(self.fdir, 'bin'), os.environ['PATH']])
```
?

Reply:
Or
```
        env_path = []
        if self.share_dir is not None:
            os.environ['PATH'] = os.pathsep.join(
                [
                    os.path.join(self.share_dir, 'bin'),
                    os.path.join(self.fdir, 'bin'),
                    os.environ['PATH']
                ]
            )
        else:
            os.environ['PATH'] = os.pathsep.join(
                [
                    os.path.join(self.fdir, 'bin'),
                    os.environ['PATH']
                ]
            )
```

Reply:
This method is likely called once on start up

Reply:
Changed

Reply:
I was thinking the first form, but yeah that's fine.